jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java

```diff
@@ -504,11 +504,19 @@ public static String safeDecodePath(String path)
      */
     public static String safeDecodePath(String path, int offset, int length)
     {
-        return decodePath(path, offset, length, URIUtil::safePathAppend);
+        return decodePath(path, offset, length, URIUtil::safePathAppend, false);
     }
 
     private static void safePathAppend(Utf8StringBuilder builder, byte b)
     {
+        // control characters
+        if (((b >= 0x00) && (b <= 0x1F)) || (b == 0x7F))
```

Comment:
There is no need to do this.   If a URI encodes control characters, then it is fine for the resulting String to include them.  They are not ambiguous and if needed we can re-encode them to get a URI.

We only need to protect reserved characters:
-  `/` as that is a path segment delimiter
- `?` as that delimits to query string... actually we might not need to protect this as we have already stripped out the query.
- `%` so we can know if character is left encoded or if it was an encoded %

Reply:
I think you might be right (again).

From the testing of this PR, this protection has broken things ...

```
org.eclipse.jetty.ee10.proxy.BalancerServletTest.testRewrittenBalancerWithEncodedURI()

Expected: a string containing "pathInfo='/test/\n'"
     but: was "requestURI='/context/mapping/test/%0A'
servletPath='/mapping'
pathInfo='/test/?'
"
```